spark-jobs/src/main/scala/za/co/absa/enceladus/common/plugin/PluginLoader.scala

```diff
@@ -46,7 +46,13 @@ class PluginLoader[+A <: Plugin:ClassTag:universe.TypeTag] {
       val key = s"$configKeyPrefix.$i"
       val factoryName = config.getString(key)
       log.info(s"Going to load a plugin factory for configuration: '$key'. Factory name: $factoryName")
-      plugins += buildPlugin(factoryName, config)
+      val plugin = buildPlugin(factoryName, config)
+      if (plugin == null) {
+        log.error(s"A NULL is returned when building a plugin: '$key'. Factory name: $factoryName")
```

Comment:
I left it as an option for a plugin developer. If the plugin is critical to the job, the factory will throw an exception. If it is not critical, it will return `null` that will be converted to Option as per Georgi suggestion.

Reply:
Fair enough